### PR TITLE
Validate SPIRV Version number when parsing binary header

### LINKED ISSUE #3831: spirv-val: version number in header not checked that it is a valid version number
An unknown/invalid value like 0x00010021 will be accepted. 
If further a tool further down the chain, i.e. spirv-cross, is more strict about the version numbers this can cause problems.

In spvBinaryHeaderGet there is a TODO about validation.

### PR DESCRIPTION
**Pull request by zoddicus**
Fixes #3831

### CODE REVIEW

File: source/binary.cpp [hunk 1]
```diff
@@ -45,6 +45,21 @@ spv_result_t spvBinaryHeaderGet(const spv_const_binary binary,
   // TODO: Validation checking?
   pHeader->magic = spvFixWord(binary->code[SPV_INDEX_MAGIC_NUMBER], endian);
   pHeader->version = spvFixWord(binary->code[SPV_INDEX_VERSION_NUMBER], endian);
+  // Assumes versions are composed of a major and minor number, where major = 1
+  // and minor {0, current spec minor}. If version 2.x or 1.1x is released this
+  // check will need to be updated.
+
+  bool is_valid_version = false;
+  for (uint32_t minor = 0;
```
Comment (alan-baker) [inline source/binary.cpp:53]: Why not just?
```
// Minimum version was 1.0 and max version is defined by SPV_VERSION.
if (pHeader->version < SPV_SPIRV_VERSION_WORD(1, 0) || pHeader->version > SPV_VERSION)
  return SPV_ERROR_INVALID_BINARY;
```

This should handle all future versions too.

**Review by alan-baker — CHANGES_REQUESTED** [review]
You seem to be missing tests that catch invalid versions

File: source/binary.cpp (hunk 1 above)
Comment (zoddicus) [reply source/binary.cpp]: That is a much more elegant solution.

**Comment by zoddicus** [comment]
> You seem to be missing tests that catch invalid versions

Added tests now

**Review by alan-baker — APPROVED** [review]

**Pull request merged**